### PR TITLE
Concordia

### PR DESCRIPTION
**Pull request by lassemoldrup**
## Purpose

Implements the basic functionality of Concordia, including Discord and Telegram bots, Discord and Telegram issuers, and a verifier.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: examples/some-verifier/src/db.rs [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+                },
+            );
+
+        statement.push_str(&columns);
+        statement.push_str(&format!(" FROM {VERIFICATIONS_TABLE}"));
+        statement.push_str(&joins);
+        statement.push_str(&format!(
+            " WHERE {}.{ID_COLUMN} = $1",
+            platform.table_name()
+        ));
```
Comment (soerenbf) [inline examples/some-verifier/src/db.rs:226]: Referring to a discussion in office regarding sql injection. Is it bad to build statements like this (besides being a bit hard to follow what the final statement will be)?

What we try to do here is to make something which works as `Platform::SUPPORTED_PLATFORMS` is extended with more members.

Comment (soerenbf) [reply examples/some-verifier/src/db.rs:226]: @abizjak

File: examples/some-verifier/frontend/config.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      "index": "5924",
+      "subindex": "0"
+    },
+    "discord": {
+      "url": "http://127.0.0.1:8081",
+      "chain": "testnet",
+      "index": "5968",
+      "subindex": "0"
+    }
+  }
```
Comment (soerenbf) [inline examples/some-verifier/frontend/config.json:16]:
```suggestion
  "discordClientId": "1127954266213056635",
  "chain": "testnet",
  "issuers": {
    "telegram": {
      "url": "http://127.0.0.1:8080",
      "index": "5924",
      "subindex": "0"
    },
    "discord": {
      "url": "http://127.0.0.1:8081",
      "index": "5968",
      "subindex": "0"
    }
  }
```
?

**Review by soerenbf — COMMENTED** [review]
Also a little bit of a self-review, but here goes 😄

File: examples/some-verifier/frontend/src/lib/types.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+export enum Platform {
+  Telegram = 'telegram',
+  Discord = 'discord',
+}
+
+export interface Issuer {
+  url: string;
+  chain: string;
```
[the remark this answers is not in the document]
Comment (abizjak) [reply examples/some-verifier/frontend/src/lib/types.ts:13]: No, it does not.

File: examples/some-verifier/frontend/index.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<!doctype html>
+<html lang="en" data-bs-theme="dark">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
```
[the remark this answers is not in the document]
Comment (abizjak) [reply examples/some-verifier/frontend/index.html]: This is removed now (by you :P)

File: examples/some-verifier/frontend/public/vite.svg
[the remark this answers is not in the document]
Comment (abizjak) [reply examples/some-verifier/frontend/public/vite.svg:1]: And this.

File: examples/some-verifier/src/db.rs (hunk 1 above)
Comment (abizjak) [reply examples/some-verifier/src/db.rs]: I revised this quite a bit. Now the statements are simpler and constructed in a more direct way.

File: examples/some-verifier/frontend/config.json (hunk 2 above)
Comment (abizjak) [reply examples/some-verifier/frontend/config.json]: This file is gone now.

File: examples/some-verifier/.gitignore
[the remark this answers is not in the document]
Comment (abizjak) [reply examples/some-verifier/.gitignore:1]: I removed it now.

File: examples/bots/telegram-bot/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+[dependencies]
+teloxide = { version = "0.12", features = ["macros"] }
+tracing = "0.1"
+tracing-subscriber = "0.3"
+tokio = { version = "1.29", features = ["rt-multi-thread", "macros"] }
+clap = { version = "4", features = ["derive", "env"] }
+reqwest = { version = "0.11", features = ["json"] }
+serde = { version = "1.0", features = ["derive"] }
+anyhow = "1.0"
+some-verifier-lib = { version = "0.1", path = "../../some-verifier-lib" }
```
[the remark this answers is not in the document]
Comment (abizjak) [reply examples/bots/telegram-bot/Cargo.toml]: Fixed.

File: examples/some-issuer/.gitignore
[the remark this answers is not in the document]
Comment (abizjak) [reply examples/some-issuer/.gitignore:1]: I agree with this so I have removed the file.

File: examples/some-verifier/frontend/src/components/Verify.tsx [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+        return;
+      }
+
+      setProofError('');
+      setOpen('2');
+    })().catch((error) => {
+      setProofError('Proof creation failed.');
+      console.error(error);
+    });
+  };
```
[the remark this answers is not in the document]
Comment (abizjak) [reply examples/some-verifier/frontend/src/components/Verify.tsx]: It looks like somebody (was it you?) changed this.

**Review by abizjak — APPROVED** [review]
Thanks for all the work in the last weeks to get this into a shape that we can launch with.

It's looking good.

4 comments are unresolved @soerenbf , but since they are minor I will leave them as such and we can figure out later.
